main.go

```diff
@@ -80,6 +81,21 @@ func realMain() {
 		showVersion   bool
 	)
 
+	// If GOMAXPROCS has not been set, restrict our thread usage
+	// so we don't grow many idle threads on large core count systems,
+	// which un-necessarily consume host PID space (and thus set an
+	// artificial max limit on the number of concurrent containers we can
+	// run)
+	if os.Getenv("GOMAXPROCS") == "" {
+		// We choose 6 as we want a couple of threads for the runtime (gc etc.)
+		// and couple of threads for our parallel user code, such as the copy
+		// code in shim.go
+		const maxthreads = 6
+		if runtime.NumCPU() > maxthreads {
+			runtime.GOMAXPROCS(6)
```

Comment:
`runtime.GOMAXPROCS(maxthreads)` maybe? :)

Same comment as for https://github.com/kata-containers/proxy/pull/118 - might be worth moving the `const` to the top for better visibility.

Reply:
s/6/maxthreads/ indeed (oops ;-)).
The const - yeah, was not sure if better at the top or here.
I may also move the code to the actual `main()`, to fix the complexity CI niggle. It's not really clear to me why we have `realMain()` ;-)